src/ui/uploader.js

```diff
@@ -64,15 +64,40 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
-    /* Updates the uploader's internal file representation with
-     * information stored in the DOM's uploader element.
+    /* Updates the uploader's internal file representation with information
+     * stored in the DOM's uploader element. If the file is an image, an attempt
+     * will be made to rotate it to match any orientation EXIF data it provides.
      * - "that": the uploader itself
      * - "fileInput": the DOM uploader element
      */
     sjrk.storyTelling.block.singleFileUploader.handleFileInputChange = function (that, fileInput) {
         var fileList = fileInput[0].files;
         var currentFile = fileList[0];
-        that.currentFile = currentFile;
+
+        if (currentFile && currentFile.type.indexOf("image") === 0) {
+            loadImage(currentFile, function (img) {
```

Comment:
Can you add code comments about the purpose of using `loadImage()` and `.toBlob()`? I understand `loadImage()` is to extract the image's EXIF data but don't know what `.toBlob()` is for. Thanks.

Reply:
`.toBlob()` is used because the call to `.loadImage()` returns a `canvas` element, but what we want is file data (a blob) that we can serialize in the form. We know it's a canvas that's being returned because setting the `orientation` option to `true` will also enable the `canvas` option in the call to `loadImage`: https://github.com/blueimp/JavaScript-Load-Image#options

Reply:
I'll add comments for sure!